filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -3489,6 +3489,14 @@ void OpenGLDriver::whenFrameComplete(const std::function<void()>& fn) noexcept {
 
 void OpenGLDriver::whenGpuCommandsComplete(const std::function<void()>& fn) noexcept {
     GLsync sync = glFenceSync(GL_SYNC_GPU_COMMANDS_COMPLETE, 0);
+    // glFenceSync can return 0 in certain situations. For example, we've seen this happen on Web
+    // when the GL context has been lost.
+    // Unfortunately, there's not much we can do here. The fn callback will never execute, which
+    // could cause issues. If we've lost the WebGL context though, Filament is most likely shutting
```

Comment:
It's very dangerous. e.g. the callback could be releasing a condition variable or memory.
I think I would prefer to do this only on Web (with a log); and assert in the other cases.

Reply:
Yeah that sounds reasonable.